pandas/tests/io/test_sql.py

```diff
@@ -1307,9 +1312,8 @@ def check(col):
         assert is_datetime64tz_dtype(col.dtype)
         assert str(col.dt.tz) == 'UTC'
         expected = sql.read_sql_table("types_test_data", self.conn)
-        tm.assert_series_equal(df.DateColWithTz,
-                               expected.DateColWithTz
-                               .astype('datetime64[ns, UTC]'))
+        # Removed ".astype('datetime64[ns, UTC]')"after GH 6415 was fixed
+        tm.assert_series_equal(df.DateColWithTz, expected.DateColWithTz)
 
         # xref #7139
         # this might or might not be converted depending on the postgres driver
```

Comment:
Can't comment on the exact location, but can you add the `assert is_datetime64tz_dtype(col.dtype)` on the line below this as well? 
(it's probably a superfluous check, but this is the place where read_sql_table is explicitly tested, so I think it is clearer to do the check here as well (above the result of `read_sql_table` is used as the expected result)

Reply:
So just to confirm, essentially(?)

```
expected = sql.read_sql_table("types_test_data", self.conn)
col = expected.DateColWithTz
assert is_datetime64tz_dtype(col.dtype)
```

Reply:
yep